addons/binding/org.openhab.binding.lgwebos/src/main/java/org/openhab/binding/lgwebos/handler/LGWebOSHandler.java

```diff
@@ -101,6 +102,15 @@ public void handleCommand(ChannelUID channelUID, Command command) {
 
     @Override
     public void initialize() {
+        if (!getConfig().containsKey(PROPERTY_DEVICE_ID)) {
+            logger.debug(
+                    "Migrating to new Thing configuration. Setting deviceID property based on ThingID {}. This is only required once, when upgrading existing LG WebOS things to binding version to 2.2.",
```

Comment:
Did you find out if the framework still creates the thing handler when it is missing the now required deviceId property on existing things? This logic would then be executed when upgrading to version 2.3 and not 2.2. :-)

Reply:
I am actually having issues right now in the case of a fresh start. I put the deviceId into the discovery result. when I add the thing from the inbox the property is null in initialize(). it does show however in paper ui under properties

Reply:
what I mean is, upon discovery it always runs through this if statement. I would expect  that the property is set in the config when I create a new thing from scratch

Reply:
OK, I found out that the value from the discovery result is in getThing().getProperties(). How does this relate to a Thing's configuration?     sorry for asking these beginner questions...

Reply:
it did 👍

Reply:
Great! :-)

When using the representation property it should also be added to the thing type definition (see [ESH documentation](https://www.eclipse.org/smarthome/documentation/development/bindings/thing-definition.html#representation-property)):

So for the WebOSTV thing type the following should be added:
```
<representation-property>deviceId</representation-property>
```

Reply:
done